action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -284,10 +308,10 @@ func (stateDB *StateDBAdapter) GetNonce(evmAddr common.Address) uint64 {
 		log.L().Error("Failed to get nonce.", zap.Error(err))
 		// stateDB.logError(err)
 	} else {
-		nonce = state.Nonce
-	}
-	if stateDB.usePendingNonce {
-		nonce++
+		nonce = state.PendingNonce()
+		if !stateDB.usePendingNonce {
+			nonce--
```

Comment:
is it possible a new account (type = 1) will go to here? then nonce would be 0-1